### PR TITLE
enhance: [2.4] add scalar filtering and vector search latency metrics 

### PR DESCRIPTION
**Pull request by foxspy**
issue: #34780 
pr: #34785

### CODE REVIEW

**Comment by foxspy** [comment]
pr: #34785

**Comment by foxspy** [comment]
rerun ut

File: internal/core/unittest/test_monitor.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#include <gtest/gtest.h>
+
+#include "monitor/monitor_c.h"
+#include "monitor/prometheus_client.h"
+
+using namespace std;
+
+vector<string>
+split(const string& str,
+      const string& delim) {  //将分割后的子字符串存储在vector中
```
Comment (czs007) [inline internal/core/unittest/test_monitor.cpp:23]: use english please.

Comment (foxspy) [reply internal/core/unittest/test_monitor.cpp:23]: Just moved from test_storage.cc...

**Pull request merged**